### PR TITLE
21187: Fixes bugs around constraining entity creation and mutating code

### PR DESCRIPTION
**Pull request by howsohazard**

### CODE REVIEW

File: src/Amalgam/evaluablenode/EvaluableNodeTreeManipulation.cpp [hunk 1]
```diff
@@ -1505,7 +1505,7 @@ void MutateImmediateNode(EvaluableNode *n, RandomStream &rs, std::vector<std::st
 EvaluableNode *EvaluableNodeTreeManipulation::MutateNode(EvaluableNode *n, MutationParameters &mp)
 {
 	if(n == nullptr)
-		return nullptr;
+		n = mp.enm->AllocNode(ENT_NULL);
```
Comment (apbassett) [inline src/Amalgam/evaluablenode/EvaluableNodeTreeManipulation.cpp:1508]: This is more for my own curiosity: why this change?

**Review by apbassett — APPROVED** [review]

File: src/Amalgam/evaluablenode/EvaluableNodeTreeManipulation.cpp (hunk 1 above)
Comment (howsohazard) [reply src/Amalgam/evaluablenode/EvaluableNodeTreeManipulation.cpp:1508]: So that it can mutate a null as opposed to only keeping it the same
